remove ppc64le architecture support

dasionov opened the pull request:
### What this PR does

This PR **removes the `ppc64le` architecture configuration completely** and also removes associated code references and handling logic related to the `ppc64le` architecture.

The `ppc64le` architecture is currently not supported or maintained in the KubeVirt project.

**References:**

- [Discussion in PR #14673](https://github.com/kubevirt/kubevirt/pull/14673#discussion_r2152401031)

---
### Release note
```release-note
remove ppc64le architecture configuration support
```

### Jira Ticket
```
https://issues.redhat.com/browse/CNV-64361
```

lyarwood commented:
@dasionov I've likely missed where this was discussed but could you include a reference in the PR or even better a tracking issue for the removal of `ppc64le` support entirely from KubeVirt?

dasionov commented:
[Change](https://github.com/kubevirt/kubevirt/compare/e0cccf88a7c3d3def724f60fb2803cee76bd932d..c9811534653ba2a609a8d8423ef2a316237cc8a1): Rebase

lyarwood commented:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/14976/pull-kubevirt-generate/1945159876569206784

@dasionov as I said I'm not sure what's generating this and if it's safe to remove so potentially something for a follow up.

dasionov commented:
> https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/14976/pull-kubevirt-generate/1945159876569206784
> 
> @dasionov as I said I'm not sure what's generating this and if it's safe to remove so potentially something for a follow up.

yes, my thoughts exactly.
Thanks!

lyarwood reviewed (commented):
/lgtm

dasionov commented on pkg/virt-controller/services/template.go:
```diff
@@ -786,9 +781,6 @@ func (t *templateService) newInitContainerRenderer(vmiSpec *v1.VirtualMachineIns
 		cpInitContainerOpts = append(cpInitContainerOpts, WithNonRoot(userId))
 	}
-	if t.IsPPC64() {
-		cpInitContainerOpts = append(cpInitContainerOpts, WithPrivileged())
```
little bit out of the scope of this PR, I would like to keep the review process as simple as possible. 
will gladly address this matter in a follow-up PR

lyarwood reviewed (commented):
/lgtm

FWIW there's also a tiny amount of code in kubevirtci to clean up:

https://github.com/kubevirt/kubevirtci/blob/543b278ad1141b91cc5f35fd64c192c4f30a142a/cluster-up/cluster/kind/common.sh#L28-L44

dasionov commented:
[Change](https://github.com/kubevirt/kubevirt/pull/14976/commits/78a95f1ecc90739857d78bf0adcf7c9a8ce4b622): address @xpivarc and @jschintag [comment](https://github.com/kubevirt/kubevirt/pull/14976#discussion_r2213084088) to remove unused `WithPrivileged` Code
PTAL :pray:

dasionov commented:
/retest-required

xpivarc commented on staging/src/kubevirt.io/api/core/v1/types.go:
```diff
@@ -2938,8 +2938,10 @@ const (
+	Amd64 *ArchSpecificConfiguration `json:"amd64,omitempty"`
+	Arm64 *ArchSpecificConfiguration `json:"arm64,omitempty"`
+
+	// Deprecated: ppc64le architecture is no longer supported.
```
Should we raise warning if set?

jschintag replied:
Shouldn't the webhook for the kubevirt CR not be handled by virt-operator? Otherwise it makes little sense. That is the one CRD that should not be handled by virt-api.

dasionov replied:
it is handled by virt-operator, I managed to find where and add a warning when the kubevirt CR gets updated
see this [commit](https://github.com/kubevirt/kubevirt/pull/14976/commits/3c44f7c709b2de4e9f52eb80a5569a82acb31480)

dasionov commented on pkg/virt-controller/services/template_test.go:
```diff
@@ -519,7 +519,6 @@ var _ = Describe("Template", func() {
 			},
 				Entry("on amd64", "amd64", "/usr/share/OVMF"),
 				Entry("on arm64", "arm64", "/usr/share/AAVMF"),
-				Entry("on ppc64le", "ppc64le", "/usr/share/OVMF"),
```
well.. who knew :wink:

0xFelix reviewed (commented):
There is still an arch defaulter for ppc64le in `pkg/virt-launcher/virtwrap/api/arch-defaulter`. Can you remove that too?

Also, `docs/build-the-builder.md` and `tests/virtctl/vnc.go` still mention ppc64le, can you update them?

dasionov commented:
> There is still an arch defaulter for ppc64le in `pkg/virt-launcher/virtwrap/api/arch-defaulter`. Can you remove that too?
> 
> Also, `docs/build-the-builder.md` and `tests/virtctl/vnc.go` still mention ppc64le, can you update them?

Rebased and fixed all the places I found with ppc64le, but didn't see any  at `tests/virtctl/vnc.go` could you PTAL :pray:

0xFelix commented:
There's a comment on L115 in `tests/virtctl/vnc.go` that mentions `ppc64le`.

dasionov commented:
/retest-required

dasionov commented:
[Change](https://github.com/kubevirt/kubevirt/compare/3ac7f12a613bd90af7e518314eaef18da7780538..ad25cd9210cb182df0d6b033c572398da20f3b45): Rebase

0xFelix reviewed (approved):
Re-lgtm

/lgtm

vladikr commented:
Looks good, thank you @dasionov 
/approve
